### PR TITLE
Cscfairmeta 465 spatial coverage

### PR DESCRIPTION
**Pull request by ismotee**
Spatial coverage added with full functionality.

Collapseable sections added.

### CODE REVIEW

File: etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/AddSpatialCoverage.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                contentLabel="Add spatial coverage modal"
+                customStyles={modalStyle}
+              >
+                <ModalContent />
+              </Modal>
+            ) : null}
+            <ButtonContainer>
+              <AddNewButton type="button" onClick={this.open}>
+                <Translate content="qvain.temporalAndSpatial.spatial.addButton" />
+              </AddNewButton>
```
[the remark this answers is not in the document]
Comment (ismotee) [reply etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/AddSpatialCoverage.jsx:42]: Yea Adding things needs some response to the User. Ux designer will look into it. The ticket added.

File: etsin_finder/frontend/js/components/qvain/general/buttons.jsx [hunk 2]
```diff
@@ -41,15 +41,15 @@ export const CancelButton = styled.button`
...
+  border: ${(props) => `solid 1px ${ props.theme.color.primary}` };
+  background-color: ${(props) => `${ props.theme.color.primary}` };
   font-size: 20px;
   font-weight: bold;
   color: #fff;
   margin-left: 20px;
   padding: 10px 25px;
   &:hover {
-    background-color: #3a813b;
+    background-color: #004d79;
```
Comment (tahme) [inline etsin_finder/frontend/js/components/qvain/general/buttons.jsx:52]: Use `theme.color.primaryDark`

File: etsin_finder/frontend/js/components/qvain/utils/select.js [hunk 3]
```diff
@@ -31,3 +33,31 @@ export const onChange = (options, lang, callback, constructFunc) => (selection)
...
+  const response = await api.get(`_search?size=100&q=*${inputValue}*`)
+  return parseRefResponse(response)
+}
+
+const refDataApi = ref =>
+  axios.create({
+    baseURL: `https://metax.csc.local/es/reference_data/${ref}/`,
+  })
+
+const parseRefResponse = res => {
```
Comment (tahme) [inline etsin_finder/frontend/js/components/qvain/utils/select.js:49]: Some references don't have translations in all languages, e.g. "Luotola (Luumäki)" has no English translation. Use `ref._source.label.en || ref._source.label.und`.

File: etsin_finder/frontend/js/components/qvain/temporalAndSpatial/index.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  const fieldProps = {
+    translations,
+    components,
+  }
+
+const TemporalAndSpatial = () => (
+  <Field {...fieldProps}>
+    <Spatial />
+  </Field>
+    )
```
Comment (tahme) [inline etsin_finder/frontend/js/components/qvain/temporalAndSpatial/index.jsx:24]:
```suggestion
)
```

File: etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/Spatials.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            onClick={() => removeSpatial(spatial.uiid)}
+            attribute={{ 'aria-label': 'qvain.general.buttons.remove' }}
+          />
+        </ButtonContainer>
+      </SpatialContainer>
+    ))
+
+    return (
+      <>{SpatialElements}</>
+    )
```
Comment (tahme) [inline etsin_finder/frontend/js/components/qvain/temporalAndSpatial/spatial/Spatials.jsx:40]: No need for jsx wrapper, you can return the array directly:
```suggestion
    return SpatialElements
```

**Pull request merged**